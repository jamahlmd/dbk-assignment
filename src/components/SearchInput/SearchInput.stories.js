import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import StoryContainer from '../StoryContainer';
import SearchInput from './index';


storiesOf('SearchInput', module)
    .add('SearchInput without remove button', () => {
        return (
            <StoryContainer>
                <SearchInput
                    value='value'
                    onChange={action('Input')}
                    onSubmit={action('Submitted')}
                >
                    <SearchInput.Input />
                    <SearchInput.RemoveButton />
                    <SearchInput.SearchButton />
                </SearchInput>
            </StoryContainer>
        )
    })
    .add('SearchInput with remove button', () => {
        return (
            <StoryContainer>
                <SearchInput
                    value='value'
                    onChange={action('Input')}
                    onSubmit={action('Submitted')}
                    handleRemove={action('Remove')}
                >
                    <SearchInput.Input />
                    <SearchInput.RemoveButton />
                    <SearchInput.SearchButton />
                </SearchInput>
            </StoryContainer>
        )
    })
    .add('SearchInput with searchbutton on the left', () => {
        return (
            <StoryContainer>
                <SearchInput
                    value='value'
                    onChange={action('Input')}
                    onSubmit={action('Submitted')}
                    handleRemove={action('Remove')}
                >
                    <SearchInput.SearchButton />
                    <SearchInput.Input />
                    <SearchInput.RemoveButton />
                </SearchInput>
            </StoryContainer>
        )
    })
    ;