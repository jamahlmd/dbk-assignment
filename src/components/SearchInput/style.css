.search-form {
    display: flex;
    border: 2px solid #e6e6e6;
    margin: 10px;
    transition: border-color .1s ease;
}

.search-form__active {
    border-color: #afafaf;
}

.search-form--input {
    border: none;
    outline: none;
    flex: 1;
    padding: 9px 0 9px 9px;
    color: #3e3e3e;
}

.search-form--button {
    border: none;
    outline: none;
    height: 33px;
    width: 33px;
    padding: 0;
    background-color: transparent;
    cursor: pointer;
    transition: background-color .3s ease
    ,opacity .1s ease;
    animation: fadein .2s;
}

@keyframes fadein {
    from { opacity: 0; }
    to   { opacity: 1; }
}

.search-form--button:hover {
    background-color: #f6f6f6;
}