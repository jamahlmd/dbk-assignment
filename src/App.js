import React, { Component, Fragment } from 'react';
import SearchInput from './components/SearchInput';
import Header from './components/Header';
import Suggestions from './components/Suggestions';

import { fetchSuggestions } from './api/suggestions';

class App extends Component {
  state = {
    query: '',
    suggestions: [],
    error: {},
  }
  onChange = (e) => this.setState({ [e.target.name]: e.target.value });

  onSubmit = () => console.log('Submit button clicked');

  handleRemove = () => this.setState({ query: '', suggestions: [], });

  handlePickSuggestion = (suggestion) =>
    this.setState({ query: suggestion }, () => this.onSubmit());

  fetchSuggestions = async () => {
    try {
      const data = await fetchSuggestions(this.state.query);
      this.setState({
        suggestions: data.suggestions.slice(0, 4),
      });
    } catch (error) {
      this.setState({ error });
    }
  }

  render() {
    return (
      <Fragment>
        <Header>
          <SearchInput
            value={this.state.query}
            onChange={this.onChange}
            onSubmit={this.onSubmit}
            onKeyUp={this.fetchSuggestions}
            handleRemove={this.handleRemove}
          >
            <SearchInput.Input />
            <SearchInput.RemoveButton />
            <SearchInput.SearchButton />
          </SearchInput>
          <Suggestions
            handlePickSuggestion={this.handlePickSuggestion}
            query={this.state.query}
            suggestions={this.state.suggestions}
          />
        </Header>
      </Fragment>
    );
  }
}

export default App;
