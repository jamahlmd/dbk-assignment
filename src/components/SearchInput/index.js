import React, { useState, useContext } from 'react';
import classnames from 'classnames';
import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import './style.css';

/* Created a compound component to give the user of this component control over the render order    * making the component more flexible 
 */

// Declare a context used to pass values down to child components
const SearchInputContext = React.createContext();

// Declare let variable to store the input ref which is used to focus the input
let SearchInputRef;

const SearchInput = ({ value, onChange, handleRemove, onSubmit, onKeyUp, children }) => {

    // Declare state variable using react Hooks
    const [isFocused, setFocus] = useState(false);

    return (
        <SearchInputContext.Provider value={{ setFocus, value, onChange, onKeyUp, handleRemove, onSubmit }}>
            <form className={classnames('search-form', {
                'search-form__active': isFocused,
            })}>
                {children}
            </form>
        </SearchInputContext.Provider>
    );
};

const Input = () => {
    const { value, onChange, setFocus, onKeyUp } = useContext(SearchInputContext);
    return (
        <input
            type="text"
            value={value}
            onChange={onChange}
            onKeyUp={() => {
                // Only add onKeyUp function if one is provided
                if (value.length > 2 && onKeyUp) onKeyUp();
            }}
            className="search-form--input"
            placeholder="Zoeken"
            name="query"
            ref={input => SearchInputRef = input}
            onFocus={() => setFocus(true)}
            onBlur={() => setFocus(false)}
        />
    );
};

SearchInput.Input = Input;

const SearchButton = () => {
    const { onSubmit } = useContext(SearchInputContext);
    return (
        <button
            onClick={(e) => {
                e.preventDefault();
                onSubmit();
            }}
            className="search-form--button">
            <FontAwesomeIcon icon={faSearch} />
        </button>
    );
};

SearchInput.SearchButton = SearchButton;

const RemoveButton = () => {
    const { handleRemove, value } = useContext(SearchInputContext);
    // If handleRemove is not defined then the remove button won't show up
    if (value.length > 0 && handleRemove) {
        return (
            <button
                onClick={(e) => {
                    if (e.target.keyCode !== 13) {
                        e.preventDefault();
                        handleRemove();
                        SearchInputRef.focus();
                    }
                }}
                type="button"
                className="search-form--button">
                <FontAwesomeIcon icon={faTimes} />
            </button>
        );
    }
    return null;
};

SearchInput.RemoveButton = RemoveButton;

SearchInput.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    handleRemove: PropTypes.func,
    onKeyUp: PropTypes.func,
};

export default SearchInput;