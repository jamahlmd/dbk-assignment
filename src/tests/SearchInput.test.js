import React from 'react';
import { mount } from 'enzyme';
import SearchInput from '../components/SearchInput';
import App from '../App';

const value = 'New Query String!';
const onSubmit = () => {};
const onChange = () => {};

it('Render SearchInput correctly', () => {
    const wrapper = mount(
        <SearchInput
            value={value}
            onSubmit={onSubmit}
            onChange={onChange}
        >
            <SearchInput.Input />
            <SearchInput.SearchButton />
            <SearchInput.RemoveButton />
        </SearchInput>
    );
    expect(wrapper).toMatchSnapshot();
});

it('Should change state.query on input change', () => {
    const name = 'query';
    const wrapper = mount(<App />);
    wrapper.find('.search-form--input').simulate('change', {
        target: {
            name,
            value,
        }
    });
    expect(wrapper.state('query')).toBe(value);
});