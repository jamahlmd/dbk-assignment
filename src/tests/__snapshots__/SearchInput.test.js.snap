// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SearchInput correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <SearchInput
    onChange={[Function]}
    onSubmit={[Function]}
    value="New Query String!"
>
    <Input />
    <SearchButton />
    <RemoveButton />
</SearchInput>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Input />,
        <SearchButton />,
        <RemoveButton />,
      ],
      "onChange": [Function],
      "onSubmit": [Function],
      "value": "New Query String!",
    },
    "ref": null,
    "rendered": Object {
      "instance": <form
        class="search-form"
>
        <input
                type="text"
                class="search-form--input"
                placeholder="Zoeken"
                name="query"
                value="New Query String!"
        />
        <button
                class="search-form--button"
        >
                <svg
                        aria-hidden="true"
                        focusable="false"
                        data-prefix="fas"
                        data-icon="search"
                        class="svg-inline--fa fa-search fa-w-16 "
                        role="img"
                        xmlns="http://www.w3.org/2000/svg"
                        viewBox="0 0 512 512"
                >
                        <path
                                fill="currentColor"
                                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                        />
                </svg>
        </button>
</form>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input />,
          <SearchButton />,
          <RemoveButton />,
        ],
        "className": "search-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": Object {
            "instance": <input
              type="text"
              class="search-form--input"
              placeholder="Zoeken"
              name="query"
              value="New Query String!"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-form--input",
              "name": "query",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onKeyUp": [Function],
              "placeholder": "Zoeken",
              "type": "text",
              "value": "New Query String!",
            },
            "ref": [Function],
            "rendered": Array [
              undefined,
            ],
            "type": "input",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": Object {
            "instance": <button
              class="search-form--button"
>
              <svg
                            aria-hidden="true"
                            focusable="false"
                            data-prefix="fas"
                            data-icon="search"
                            class="svg-inline--fa fa-search fa-w-16 "
                            role="img"
                            xmlns="http://www.w3.org/2000/svg"
                            viewBox="0 0 512 512"
              >
                            <path
                                          fill="currentColor"
                                          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                            />
              </svg>
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                                Object {
                                                "icon": Array [
                                                  512,
                                                  512,
                                                  Array [],
                                                  "f002",
                                                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                                                ],
                                                "iconName": "search",
                                                "prefix": "fas",
                                              }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
/>,
              "className": "search-form--button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": <svg
                    aria-hidden="true"
                    focusable="false"
                    data-prefix="fas"
                    data-icon="search"
                    class="svg-inline--fa fa-search fa-w-16 "
                    role="img"
                    xmlns="http://www.w3.org/2000/svg"
                    viewBox="0 0 512 512"
>
                    <path
                                        fill="currentColor"
                                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    />
</svg>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      style={Object {}}
/>,
                    "className": "svg-inline--fa fa-search fa-w-16 ",
                    "data-icon": "search",
                    "data-prefix": "fas",
                    "focusable": "false",
                    "role": "img",
                    "style": Object {},
                    "viewBox": "0 0 512 512",
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <path
                        fill="currentColor"
                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
/>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "d": "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                        "fill": "currentColor",
                        "style": Object {},
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "path",
                    },
                  ],
                  "type": "svg",
                },
                "type": [Function],
              },
            ],
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input />,
          <SearchButton />,
          <RemoveButton />,
        ],
        "onChange": [Function],
        "onSubmit": [Function],
        "value": "New Query String!",
      },
      "ref": null,
      "rendered": Object {
        "instance": <form
          class="search-form"
>
          <input
                    type="text"
                    class="search-form--input"
                    placeholder="Zoeken"
                    name="query"
                    value="New Query String!"
          />
          <button
                    class="search-form--button"
          >
                    <svg
                              aria-hidden="true"
                              focusable="false"
                              data-prefix="fas"
                              data-icon="search"
                              class="svg-inline--fa fa-search fa-w-16 "
                              role="img"
                              xmlns="http://www.w3.org/2000/svg"
                              viewBox="0 0 512 512"
                    >
                              <path
                                        fill="currentColor"
                                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                              />
                    </svg>
          </button>
</form>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input />,
            <SearchButton />,
            <RemoveButton />,
          ],
          "className": "search-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": Object {
              "instance": <input
                type="text"
                class="search-form--input"
                placeholder="Zoeken"
                name="query"
                value="New Query String!"
/>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "search-form--input",
                "name": "query",
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "onKeyUp": [Function],
                "placeholder": "Zoeken",
                "type": "text",
                "value": "New Query String!",
              },
              "ref": [Function],
              "rendered": Array [
                undefined,
              ],
              "type": "input",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": Object {
              "instance": <button
                class="search-form--button"
>
                <svg
                                aria-hidden="true"
                                focusable="false"
                                data-prefix="fas"
                                data-icon="search"
                                class="svg-inline--fa fa-search fa-w-16 "
                                role="img"
                                xmlns="http://www.w3.org/2000/svg"
                                viewBox="0 0 512 512"
                >
                                <path
                                                fill="currentColor"
                                                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                                />
                </svg>
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                                    Object {
                                                      "icon": Array [
                                                        512,
                                                        512,
                                                        Array [],
                                                        "f002",
                                                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                                                      ],
                                                      "iconName": "search",
                                                      "prefix": "fas",
                                                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
/>,
                "className": "search-form--button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    },
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": <svg
                      aria-hidden="true"
                      focusable="false"
                      data-prefix="fas"
                      data-icon="search"
                      class="svg-inline--fa fa-search fa-w-16 "
                      role="img"
                      xmlns="http://www.w3.org/2000/svg"
                      viewBox="0 0 512 512"
>
                      <path
                                            fill="currentColor"
                                            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      />
</svg>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": <path
                        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                        fill="currentColor"
                        style={Object {}}
/>,
                      "className": "svg-inline--fa fa-search fa-w-16 ",
                      "data-icon": "search",
                      "data-prefix": "fas",
                      "focusable": "false",
                      "role": "img",
                      "style": Object {},
                      "viewBox": "0 0 512 512",
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <path
                          fill="currentColor"
                          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
/>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "d": "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                          "fill": "currentColor",
                          "style": Object {},
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "path",
                      },
                    ],
                    "type": "svg",
                  },
                  "type": [Function],
                },
              ],
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
